# Praktikum Pertemuan 2 Data Science

Nama: Rangga Saputra

NIM: 250401020034

#Step Awal

Melakukan import library yang dibutuhkan dan juga load datasetnya. Dalam hal ini librarynya adalah **pandas dan numpy**, serta datasetnya **titanic**

In [45]:
# STEP 1: Import Libraries
import pandas as pd
import numpy as np

# STEP 2: Load Dataset Titanic
url = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(url)

print("=" * 60)
print("DATASET BERHASIL DIMUAT")
print("=" * 60)
print(df.head())



DATASET BERHASIL DIMUAT
   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.

Penjelasan:

- Library **pandas (ps)** digunakan untuk untuk mengelola data tabel (spreadsheet)
- Library **numpy (np)** digunakan untuk perhitungan matematis dengan array
- **df = pd.read_csv(url)** ini seperti membuka file Excel dari internet dan menyimpannya di RAM komputer

---



#Q1. Berapa jumlah baris dan kolom pada dataset Titanic? Kolom apa saja yang tersedia?

In [46]:
# ============================================================
# Q1. Berapa jumlah baris dan kolom pada dataset Titanic? Kolom apa saja yang tersedia?
# ============================================================
jumlah_baris, jumlah_kolom = df.shape
print(f"\nJumlah baris: {jumlah_baris}")
print(f"Jumlah kolom: {jumlah_kolom}")
print(f"\nDaftar kolom yang tersedia:")
for i, col in enumerate(df.columns, 1):
    print(f"  {i}. {col}")


Jumlah baris: 891
Jumlah kolom: 12

Daftar kolom yang tersedia:
  1. PassengerId
  2. Survived
  3. Pclass
  4. Name
  5. Sex
  6. Age
  7. SibSp
  8. Parch
  9. Ticket
  10. Fare
  11. Cabin
  12. Embarked


Penjelasan: Dari hasil ini ditemukan 891 baris, 12 kolom pada dataset Titanic

---



#Q2. Kolom mana saja yang memiliki missing values? Berapa jumlah dan proporsinya (%) per kolom?

In [47]:
# ============================================================
# Q2. Kolom mana saja yang memiliki missing values? Berapa jumlah dan proporsinya (%) per kolom?
# ============================================================
missing_info = pd.DataFrame({
    'Kolom': df.columns,
    'Jumlah_Missing': df.isnull().sum(),
    'Proporsi_%': (df.isnull().sum() / len(df) * 100).round(2)
})

missing_info = missing_info[missing_info['Jumlah_Missing'] > 0].sort_values('Jumlah_Missing', ascending=False)

if len(missing_info) > 0:
    print("\nKolom dengan missing values:")
    print(missing_info.to_string(index=False))
else:
    print("\nTidak ada missing values")


Kolom dengan missing values:
   Kolom  Jumlah_Missing  Proporsi_%
   Cabin             687       77.10
     Age             177       19.87
Embarked               2        0.22


Penjelasan:

Kolom dengan missing value adalah "Age, Cabin, dan Embarked".

- Kolom "Age" missing valuenya 177 dan propostinya 19.87%
- Kolom "Cabin" missing valuenya 687 dan propostinya 77.10%
- kolom "Embarked" missing valuenya 2 dan propostinya 0.22%

---



#Q3. Berapa persen penumpang yang selamat (Survived=1)? Gunakan operasi NumPy atau Pandas.

In [48]:
# ============================================================
# Q3. Berapa persen penumpang yang selamat (Survived=1)? Gunakan operasi NumPy atau Pandas.
# ============================================================
# Metode 1: Menggunakan Pandas
persen_selamat_pd = (df['Survived'].sum() / len(df)) * 100
jumlah_selamat_pd = df['Survived'].sum()
jumlah_tidak_selamat = len(df) - jumlah_selamat_pd

# Metode 2: Menggunakan NumPy
survived_array = np.array(df['Survived'])
persen_selamat_np = (np.sum(survived_array) / len(survived_array)) * 100

print(f"\nJumlah penumpang selamat: {int(jumlah_selamat_pd)}")
print(f"Jumlah penumpang tidak selamat: {int(jumlah_tidak_selamat)}")
print(f"Persentase selamat (Pandas): {persen_selamat_pd:.2f}%")
print(f"Persentase selamat (NumPy): {persen_selamat_np:.2f}%")

# Visualisasi tabel
survival_count = df['Survived'].value_counts()
print(f"\nRingkasan:")
print(f"  Tidak selamat: {survival_count.get(0, 0)} penumpang")
print(f"  Selamat: {survival_count.get(1, 0)} penumpang")


Jumlah penumpang selamat: 342
Jumlah penumpang tidak selamat: 549
Persentase selamat (Pandas): 38.38%
Persentase selamat (NumPy): 38.38%

Ringkasan:
  Tidak selamat: 549 penumpang
  Selamat: 342 penumpang


Penjelasan:

- Penumpang selamat sebanyak 38.38% (342 dari 891 orang)
- Sedangkan yang tidak selamat sebanyak 549 orang

---



#Q4. Filter penumpang wanita dari kelas 1 (Pclass=1). Berapa jumlahnya dan berapa rata-rata usianya?

In [49]:
# ============================================================
# Q4. Filter penumpang wanita dari kelas 1 (Pclass=1). Berapa jumlahnya dan berapa rata-rata usianya?
# ============================================================
wanita_kelas1 = df[(df['Sex'] == 'female') & (df['Pclass'] == 1)]
jumlah = len(wanita_kelas1)
rata_usia = wanita_kelas1['Age'].mean()

print(f"\nJumlah penumpang wanita kelas 1: {jumlah}")
print(f"Rata-rata usia: {rata_usia:.2f} tahun")
print(f"\nDetail penumpang wanita kelas 1:")
print(wanita_kelas1[['Name', 'Age', 'Sex', 'Pclass', 'Survived']].to_string(index=False))


Jumlah penumpang wanita kelas 1: 94
Rata-rata usia: 34.61 tahun

Detail penumpang wanita kelas 1:
                                                                              Name  Age    Sex  Pclass  Survived
                               Cumings, Mrs. John Bradley (Florence Briggs Thayer) 38.0 female       1         1
                                      Futrelle, Mrs. Jacques Heath (Lily May Peel) 35.0 female       1         1
                                                          Bonnell, Miss. Elizabeth 58.0 female       1         1
                                    Spencer, Mrs. William Augustus (Marie Eugenie)  NaN female       1         1
                                          Harper, Mrs. Henry Sleeper (Myna Haxtun) 49.0 female       1         1
                                                               Icard, Miss. Amelie 38.0 female       1         1
                                                        Fortune, Miss. Mabel Helen 23.0 female       1        

Penjelasan:
- Jumlah penumpang wanita kelas 1: 94
- Rata-rata usia: 34.61 tahun

---



#Q5. Gunakan groupby untuk menghitung tingkat keselamatan (mean Survived) per kelas penumpang. Kelas mana yang paling tinggi?

In [50]:
# ============================================================
# Q5. Gunakan groupby untuk menghitung tingkat keselamatan (mean Survived) per kelas penumpang. Kelas mana yang paling tinggi?
# ============================================================
tingkat_selamat = df.groupby('Pclass')['Survived'].agg([
    ('Jumlah_Penumpang', 'count'),
    ('Jumlah_Selamat', 'sum'),
    ('Tingkat_Selamat_%', lambda x: (x.sum() / len(x)) * 100)
]).round(2)

print("\n")
print(tingkat_selamat)

kelas_tertinggi = df.groupby('Pclass')['Survived'].mean().idxmax()
nilai_tertinggi = df.groupby('Pclass')['Survived'].mean().max()

print(f"\nKelas dengan tingkat keselamatan tertinggi: Kelas {int(kelas_tertinggi)}")
print(f"Tingkat keselamatan: {nilai_tertinggi * 100:.2f}%")

# Analisis tambahan: Perbandingan per kelas dan gender
print("\n--- Analisis Tambahan: Tingkat Keselamatan per Kelas dan Gender ---")
analisis_detail = df.groupby(['Pclass', 'Sex'])['Survived'].agg([
    ('Jumlah', 'count'),
    ('Selamat', 'sum'),
    ('Tingkat_%', lambda x: (x.sum() / len(x)) * 100)
]).round(2)
print(analisis_detail)



        Jumlah_Penumpang  Jumlah_Selamat  Tingkat_Selamat_%
Pclass                                                     
1                    216             136              62.96
2                    184              87              47.28
3                    491             119              24.24

Kelas dengan tingkat keselamatan tertinggi: Kelas 1
Tingkat keselamatan: 62.96%

--- Analisis Tambahan: Tingkat Keselamatan per Kelas dan Gender ---
               Jumlah  Selamat  Tingkat_%
Pclass Sex                               
1      female      94       91      96.81
       male       122       45      36.89
2      female      76       70      92.11
       male       108       17      15.74
3      female     144       72      50.00
       male       347       47      13.54


Penjelasan:
- Kelas 1: 62.96%
- Kelas 2: 47.28%
- Kelas 3: 24.24%

Jadi tingkat keselamatan yang paling tinggi adalah **Kelas 1**

---



#Q6. Buat analisis tambahan berdasarkan variabel lain yang Anda anggap menarik. Sertakan interpretasi singkat.

In [51]:
# ============================================================
# Q6.Buat analisis tambahan berdasarkan variabel lain yang Anda anggap menarik. Sertakan interpretasi singkat.
# ============================================================
# PENGARUH USIA TERHADAP KESELAMATAN

# Membuat kategori usia
df['Kelompok_Usia'] = pd.cut(df['Age'],
                               bins=[0, 12, 18, 35, 50, 100],
                               labels=['Anak (0-12)', 'Remaja (13-18)',
                                      'Dewasa (19-35)', 'Dewasa Tua (36-50)',
                                      'Lansia (>50)'])

analisis_usia = df.groupby('Kelompok_Usia', observed=True)['Survived'].agg([
    ('Jumlah', 'count'),
    ('Selamat', 'sum'),
    ('Tingkat_%', lambda x: (x.sum() / len(x)) * 100)
]).round(2)

print("\nTingkat keselamatan berdasarkan kelompok usia:")
print(analisis_usia)

print("\n--- Interpretasi Q6 ---")
print("""
Dari analisis berdasarkan kelompok usia, dapat dilihat bahwa:
- Anak-anak (usia 0-12 tahun) memiliki tingkat keselamatan tertinggi
- Dewasa muda (usia 19-35 tahun) memiliki tingkat keselamatan terendah
- Ini konsisten dengan kebijakan "women and children first" pada saat itu
- Faktor usia signifikan mempengaruhi peluang keselamatan di Titanic
""")

# Analisis korelasi gender
print("\n--- Analisis Tambahan: Pengaruh Gender ---")
analisis_gender = df.groupby('Sex')['Survived'].agg([
    ('Jumlah', 'count'),
    ('Selamat', 'sum'),
    ('Tingkat_%', lambda x: (x.sum() / len(x)) * 100)
]).round(2)

print(analisis_gender)
print("\nWanita memiliki peluang keselamatan yang jauh lebih tinggi")
print("dibanding pria (konsisten dengan kebijakan 'women and children first')")

# Statistik deskriptif lengkap
print("\n" + "=" * 60)
print("STATISTIK DESKRIPTIF DATASET TITANIC")
print("=" * 60)
print(df.describe().round(2))


Tingkat keselamatan berdasarkan kelompok usia:
                    Jumlah  Selamat  Tingkat_%
Kelompok_Usia                                 
Anak (0-12)             69       40      57.97
Remaja (13-18)          70       30      42.86
Dewasa (19-35)         358      137      38.27
Dewasa Tua (36-50)     153       61      39.87
Lansia (>50)            64       22      34.38

--- Interpretasi Q6 ---

Dari analisis berdasarkan kelompok usia, dapat dilihat bahwa:
- Anak-anak (usia 0-12 tahun) memiliki tingkat keselamatan tertinggi
- Dewasa muda (usia 19-35 tahun) memiliki tingkat keselamatan terendah
- Ini konsisten dengan kebijakan "women and children first" pada saat itu
- Faktor usia signifikan mempengaruhi peluang keselamatan di Titanic


--- Analisis Tambahan: Pengaruh Gender ---
        Jumlah  Selamat  Tingkat_%
Sex                               
female     314      233      74.20
male       577      109      18.89

Wanita memiliki peluang keselamatan yang jauh lebih tinggi
dibandi

Penjelasan:
- Gender paling signifikan (74.20% wanita dan 18.89% pria)
- Artinya wanita memiliki peluang keselamatan yang jauh lebih tinggi
dibanding pria (konsisten dengan kebijakan 'women and children first')


---

